goth/runner/proxy.py

```diff
@@ -69,21 +65,20 @@ def start(self):
         self._server_ready.wait()
 
     async def stop(self):
-        """Start the proxy monitor and thread."""
-        if not self._loop:
-            raise RuntimeError("Event loop is not set")
-        await self.monitor.stop()
+        """Stop the proxy thread and the monitor."""
+        self._proxy_thread.stop(StopThreadException)
         self._proxy_thread.join()
+        await self.monitor.stop()
```

Comment:
I'd prefer to stop the thread first, then the monitor, to prevent occasional errors like the one in https://github.com/golemfactory/goth/issues/468, caused by a thread that supplies events trying to add an event to a monitor that is already stopped.